src/bin/seed-database.rs

```diff
@@ -0,0 +1,207 @@
+use std::fs;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+use std::process::Command;
+
+use discv5::enr::{CombinedKey, EnrBuilder};
+use rand::{distributions::Alphanumeric, Rng};
+use rlp::{Decodable, DecoderError};
+use serde_json::Value;
+use structopt::StructOpt;
+
+use trin_core::portalnet::storage::PortalStorage;
+use trin_core::portalnet::types::content_key::MockContentKey;
+use trin_core::types::header::Header;
+use trin_core::utils::db::get_data_dir;
+use trin_history::content_key::{BlockHeader, HistoryContentKey};
+
+// This script is used to seed testnet nodes with data from mainnetMM data dump
+// https://www.dropbox.com/s/y5n36ztppltgs7x/mainnetMM.zip?dl=0
+// cargo run --bin seed-database -- --private-key 0000000000000000000000000000000000000000000000000000000000000001 --data-folder ./mainnetMM
+
+// For every 1 kb of data we store (key + value), RocksDB tends to grow by this many kb on disk...
+// ...but this is a crude empirical estimation that works mainly with default value data size of 32
+const DATA_OVERHEAD: f64 = 1.1783;
+const SIZE_OF_KEYS: u32 = 32;
+
+pub fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let generator_config = GeneratorConfig::from_args();
+
+    let mut encoded = hex::decode(&generator_config.private_key).unwrap();
+    let enr_key = CombinedKey::secp256k1_from_bytes(&mut encoded).unwrap();
+    let enr = {
+        let mut builder = EnrBuilder::new("v4");
+        builder.build(&enr_key).unwrap()
+    };
+    let node_id = enr.node_id();
+    let data_dir_path = get_data_dir(node_id);
+    println!("Storing data for NodeID: {:02X?}", node_id);
+    println!("ENR: {:?}", enr);
+    println!("DB Path: {:?}", data_dir_path);
+
+    if generator_config.overwrite {
+        let _ = Command::new("rm")
+            .arg("-rf")
+            .arg(data_dir_path)
+            .output()
+            .expect("Failed to overwrite DB.");
+    }
+
+    let num_kilobytes = generator_config.kb;
+
+    let storage_config = PortalStorage::setup_config(node_id, num_kilobytes)?;
+    let storage = PortalStorage::new(storage_config)?;
+
+    match generator_config.data_folder {
+        Some(data_folder) => load_mainnet_data(storage, data_folder),
+        None => load_random_data(storage, generator_config),
+    }
+
+    Ok(())
+}
+
+/// Type for data format stored in mainnetMM data dump
+#[derive(Debug, Clone)]
+pub struct BlockRlp {
+    pub header: Option<Header>,
+}
+
+impl Decodable for BlockRlp {
+    fn decode(rlp: &rlp::Rlp) -> Result<Self, DecoderError> {
+        let header = match Header::decode_rlp(rlp) {
```

Comment:
We can use here [result.ok()](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok):

```rust
let header = Header::decode_rlp(rlp).ok()
```